src/feedback.coffee

```diff
@@ -1,120 +1,144 @@
         if match.regex_name == 'recent_year'
-          warning: "Recent years are easy to guess"
-          suggestions: [
-            'Avoid recent years'
-            'Avoid years that are associated with you'
-          ]
+          warning: 'recent_years_are_easy'
+          suggestions: ['avoid_recent_years', 'avoid_associated_years']
 
       when 'date'
-        warning: "Dates are often easy to guess"
-        suggestions: [
-          'Avoid dates and years that are associated with you'
-        ]
+        warning: 'dates_are_easy'
+        suggestions: ['avoid_associated_dates_and_years']
 
   get_dictionary_match_feedback: (match, is_sole_match) ->
     warning = if match.dictionary_name == 'passwords'
       if is_sole_match and not match.l33t and not match.reversed
         if match.rank <= 10
-          'This is a top-10 common password'
+          'top10_common_password'
         else if match.rank <= 100
-          'This is a top-100 common password'
+          'top100_common_password'
         else
-          'This is a very common password'
+          'very_common_password'
       else if match.guesses_log10 <= 4
-        'This is similar to a commonly used password'
+        'similar_to_common_password'
     else if match.dictionary_name == 'english_wikipedia'
       if is_sole_match
-        'A word by itself is easy to guess'
+        'a_word_is_easy'
     else if match.dictionary_name in ['surnames', 'male_names', 'female_names']
       if is_sole_match
-        'Names and surnames by themselves are easy to guess'
+        'names_are_easy'
       else
-        'Common names and surnames are easy to guess'
-    else
-      ''
+        'common_names_are_easy'
 
     suggestions = []
     word = match.token
     if word.match(scoring.START_UPPER)
-      suggestions.push "Capitalization doesn't help very much"
+      suggestions.push 'capitalization_doesnt_help'
     else if word.match(scoring.ALL_UPPER) and word.toLowerCase() != word
-      suggestions.push "All-uppercase is almost as easy to guess as all-lowercase"
+      suggestions.push 'all_uppercase_doesnt_help'
 
     if match.reversed and match.token.length >= 4
-      suggestions.push "Reversed words aren't much harder to guess"
+      suggestions.push 'reverse_doesnt_help'
     if match.l33t
-      suggestions.push "Predictable substitutions like '@' instead of 'a' don't help very much"
+      suggestions.push 'substitution_doesnt_help'
 
     result =
       warning: warning
       suggestions: suggestions
     result
 
+  get_message: (key) ->
+    if @custom_messages? and key of @custom_messages
+        @custom_messages[key] or ''
+    else if @messages[key]?
+      @messages[key]
+    else
+      throw new Error("unknown message: #{key}")
+
+  build_feedback: (warning_key = null, suggestion_keys = []) ->
+    suggestions = []
+    for suggestion_key in suggestion_keys
+      message = @get_message(suggestion_key)
+      suggestions.push message if message?
```

Comment:
Please add message to suggestions only if not empty:
`suggestions.push message if message? and message != ''`

Array of suggestions with empty string like `[ '' ]` or `[ '', 'Avoid sequences']` is complication for displaying feedback. Empty string occurs if message is present in custom messages and has a falsy value.